src/anyio/pytest_plugin.py

```diff
@@ -121,7 +127,12 @@ def run_with_hypothesis(**kwargs: Any) -> None:
             funcargs = pyfuncitem.funcargs
             testargs = {arg: funcargs[arg] for arg in pyfuncitem._fixtureinfo.argnames}
             with get_runner(backend_name, backend_options) as runner:
-                runner.run_test(pyfuncitem.obj, testargs)
+                try:
+                    runner.run_test(pyfuncitem.obj, testargs)
+                except ExceptionGroup as excgrp:
+                    for exc in iterate_exceptions(excgrp):
+                        if isinstance(exc, (Exit, KeyboardInterrupt, SystemExit)):
```

Comment:
Technically I feel this is a pytest bug and should be fixed there

Reply:
Eventually, yes, but for the time being I feel this is a big quality-of-life improvement to the AnyIO testing experience.

Reply:
Yeah definitely